File: SDKs/osf.io/2.0/python/src/sdk/models/operations/preprints_partial_update.py
from dataclasses import dataclass, field
from datetime import date, datetime
from marshmallow import fields
import dateutil.parser
from typing import Any


@dataclass
class PreprintsPartialUpdatePathParams:
    preprint_id: str = field(metadata={'path_param': { 'field_name': 'preprint_id', 'style': 'simple', 'explode': False }})
    

@dataclass
class PreprintsPartialUpdateRequest:
    path_params: PreprintsPartialUpdatePathParams = field()
    request: dict[str, Any] = field(metadata={'request': { 'media_type': 'application/json' }})
    

@dataclass
class PreprintsPartialUpdateResponse:
    content_type: str = field()
    status_code: int = field()
    

File: SDKs/osf.io/2.0/typescript/src/sdk/models/operations/preprintspartialupdate.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



export class PreprintsPartialUpdatePathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=preprint_id" })
  preprintId: string;
}


export class PreprintsPartialUpdateRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: PreprintsPartialUpdatePathParams;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request: Map<string, any>;
}


export class PreprintsPartialUpdateResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}
